pkg/vpcmodel/nodesExplainability.go

```diff
@@ -56,66 +156,141 @@ func (c *VPCConfig) getFiltersEnablingRulesBetweenNodesPerDirectionAndLayer(
+	}
+	for _, configNode := range c.Nodes {
+		if !configNode.IsExternal() {
+			continue
+		}
+		configNodeIPBlock := common.NewIPBlockFromCidr(configNode.Cidr())
+		if nodeIPBlock.ContainedIn(configNodeIPBlock) {
+			return configNode, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find containing config node for %v", node.Name())
+}
+
+// todo: tmp prints each separately; will need to group.
+func (explanationStruct *explainStruct) String(c *VPCConfig) (string, error) {
+	resStr := ""
+	for _, rulesSrcDst := range *explanationStruct {
+		src := rulesSrcDst.src
+		dst := rulesSrcDst.dst
+		rules := rulesSrcDst.rules
+		needEgress := src.IsInternal()
+		needIngress := dst.IsInternal()
+		noIngressRules := len(rules.ingressRules) == 0 && needIngress
+		noEgressRules := len(rules.egressRules) == 0 && needEgress
+		egressRulesStr := fmt.Sprintf("Egress Rules:\n~~~~~~~~~~~~~\n%v", rules.egressRules.string(c))
+		ingressRulesStr := fmt.Sprintf("Ingress Rules:\n~~~~~~~~~~~~~~\n%v", rules.ingressRules.string(c))
+		noConnection := fmt.Sprintf("No connection between %v and %v;", src.Name(), dst.Name())
+		switch {
+		case noIngressRules && noEgressRules:
+			resStr += fmt.Sprintf("%v connection blocked both by ingress and egress\n", noConnection)
+		case noIngressRules:
+			resStr += fmt.Sprintf("%v connection blocked by ingress\n", noConnection)
+			if needEgress {
+				resStr += egressRulesStr
+			}
+		case noEgressRules:
+			resStr += fmt.Sprintf("%v connection blocked by egress\n", noConnection)
+			if needIngress {
+				resStr += ingressRulesStr
+			}
+		default: // there is a connection
+			resStr = fmt.Sprintf("The following connection exists between %v and %v: %v; its enabled by\n", src.Name(), dst.Name(),
+				rulesSrcDst.conn.String())
+			if needEgress {
+				resStr += egressRulesStr
+			}
+			if needIngress {
+				resStr += ingressRulesStr
+			}
 		}
-		conn, ok := connectivity.AllowedConnsCombined[src][dst]
-		if !ok {
-			return "", fmt.Errorf("error: there is a connection between %v and %v, but connection computation failed",
-				src.Name(), dst.Name())
+		// todo: not adding newlines in between since this will be replaced by grouping in the next PR
+	}
+	return resStr, nil
+}
+
+// todo: connectivity is computed for the entire network, even though we need only for specific src, dst pairs
+// this is seems the time spent here should be neglectable, not worth the effort of adding dedicated code.
+func (explanationStruct *explainStruct) computeConnections(c *VPCConfig) error {
+	connectivity, err := c.GetVPCNetworkConnectivity(false) // computes connectivity
+	if err != nil {
+		return err
+	}
+	for _, rulesSrcDst := range *explanationStruct {
+		// is there a connection?
+		if (len(rulesSrcDst.rules.egressRules) > 0 || rulesSrcDst.src.IsExternal()) && // egress enabled or not needed
+			(len(rulesSrcDst.rules.ingressRules) > 0 || rulesSrcDst.dst.IsExternal()) { // ingress enabled or not needed
```

Comment:
IsExternal()  is a method of DrawioResourceIntf interface, and not related to explainability
please use the Node interface methods:
type Node interface {
	VPCResourceIntf
	Cidr() string
	IsInternal() bool
	IsPublicInternet() bool
}

Reply:
done